Mathlib/Data/Sym/Sym2.lean

```diff
@@ -589,20 +592,25 @@ theorem fromRel_ne : fromRel (fun (_ _ : α) z => z.symm : Symmetric Ne) = {z |
   ext z; exact z.ind (by simp)
 
 lemma diagSet_eq_fromRel_eq : diagSet = fromRel (α := α) eq_equivalence.symmetric := by
-  ext z
-  exact z.ind fun _ _ ↦ mem_diagSet_iff_eq
+  ext ⟨a, b⟩; simp
 
 lemma diagSet_compl_eq_fromRel_ne : diagSetᶜ = fromRel (α := α) (r := Ne) (fun _ _ ↦ Ne.symm) := by
-  ext z
-  exact z.ind fun _ _ ↦ mem_diagSet_iff_eq.not
+  ext ⟨a, b⟩; simp
+
+@[simp] lemma diagSet_subset_fromRel (hr : Symmetric r) : diagSet ⊆ fromRel hr ↔ Reflexive r := by
+  simp [Set.subset_def, Sym2.forall, Reflexive]
+
+@[simp] lemma disjoint_diagSet_fromRel (hr : Symmetric r) :
+    Disjoint diagSet (fromRel hr) ↔ Irreflexive r := by
```

Comment:
The complement of `diagSet` [appears in 4 lemmas](https://github.com/search?q=repo%3Aleanprover-community%2Fmathlib4+diagSet%E1%B6%9C+path%3A%2F%5EMathlib%5C%2FCombinatorics%5C%2FSimpleGraph%5C%2F%2F&type=code) so we should either keep both the `Disjoint` version and the subset-compl version, or change those lemmas.
(or let them deal with it with `Disjoint.subset_compl_left`, but I prefer making the `diagSet` API easy to use)